pkg/gui/commits_panel.go

```diff
@@ -448,3 +461,56 @@ func (gui *Gui) handleSwitchToCommitFilesPanel(g *gocui.Gui, v *gocui.View) erro
 
 	return gui.switchFocus(g, v, gui.getCommitFilesView())
 }
+
+func (gui *Gui) handleToggleDiffCommit(g *gocui.Gui, v *gocui.View) error {
+	selectLimit := 2
+
+	// get selected commit
+	commit := gui.getSelectedCommit(g)
+	if commit == nil {
+		return gui.renderString(g, "main", gui.Tr.SLocalize("NoCommitsThisBranch"))
+	}
+
+	// if already selected commit delete
+	if _, has := gui.State.DiffEntries[commit.Sha]; has {
+		delete(gui.State.DiffEntries, commit.Sha)
+		gui.setDiffMode()
+	} else {
+		if len(gui.State.DiffEntries) == selectLimit {
+			return gui.createErrorPanel(gui.g, "you have already selected two commit, please deselect one of two")
```

Comment:
I think we should change the logic so that if two have been selected, we just unselect the first commit that you selected. That way if you switch branches and want to select two new commits, you don't need to go back to the original branch to deslect the old ones.

Reply:
That means we'll need to change the way we store these. I think an array of two elements makes sense, and we just remove the first element if we want to add a third one

Reply:
>That means we'll need to change the way we store these. I think an array of two elements makes sense, and we just remove the first element if we want to add a third one

I fixed it exactly